examples/systemcalls/systemcalls.c

```diff
@@ -90,10 +160,71 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
  *   redirect standard out to a file specified by outputfile.
  *   The rest of the behaviour is same as do_exec()
- *
+ *   
 */
 
-    va_end(args);
+    
+    int status;
+    pid_t pid;
 
+    int fd = open(outputfile, O_WRONLY | O_CREAT | O_TRUNC, FILE_PERMISSION);
+    if(fd == FAIL)
+    {
+
+        perror("ERROR: File not created/opened");
+        return false;
+    }
+
+    pid = fork();
+    
+    if (pid == FAIL)  // Error check of child process
+    {
+
+    	perror("ERROR: Fork failed");
+    	
+    	return false;
+    }
+    
+    else if (pid == 0) // Child process
+    { 
+	printf("Child created successfully with pid %d", pid);
+	
+        if(dup2(fd,1) < 0)
+        {
+       
+            perror("ERROR: Dup2 error");
+            return false;
+        }
+        
+        close(fd);
+        
+        execv(command[0],command);
+
+    	perror("ERROR: exev failed");   // This line executes if exec fails
+    	
+    	exit(EXIT_FAILURE);
+    }
+    else
+    {
+        close(fd);
+
+    	printf("Parent process with pid %d", pid);
+    	
+    	if(waitpid(pid,&status,0) == FAIL)    // Inside parent process
+    	{
+            perror("ERROR: Waitpid failed");
+            return false;
```

Comment:
Missing cleanup in these return paths.
Also see the discussion in class today about a way to avoid duplication in these two functions.